Dockerfiles/midstream/unit_tests.py

```diff
@@ -3,78 +3,72 @@
 import os
 import subprocess
 
+SCRIPT_DIR = "/operator-test-playbooks/Dockerfiles/midstream/"
 
 class Testing(unittest.TestCase):
+
+    def setUp(self):
+        self.env = os.environ.copy()
+        self.env["ANSIBLE_CONFIG"] = SCRIPT_DIR
+        self.env["ANSIBLE_LOCAL_TEMP"] = "/tmp/"
+        self.exec_cmd = "run_tests.py"
+
     # Running the container with correctly set environment variables, all tasks from run_tests.py should pass
     def test_default_positive(self):
-        exec_cmd = """
-        export IMAGE_TO_TEST=quay.io/cvpops/test-operator:v1.0-16
-        /run_tests.py
-        """
-        result = subprocess.run(exec_cmd, shell=True)
-        self.assertFalse(os.path.exists(".errormessage"), "File .errormessage should not be present!")
+        self.env["IMAGE_TO_TEST"] = "quay.io/cvpops/test-operator:v1.0-16"
+        result = subprocess.run(self.exec_cmd, 
+                                shell=True,
+                                env=self.env,
+                                cwd=SCRIPT_DIR)
         self.assertEqual(0, result.returncode)
+        
 
     # Running the container with correctly set environment variables, default channel missing, all tasks should pass
     def test_positive_missing_default_channel(self):
-        exec_cmd = """
-        export IMAGE_TO_TEST=quay.io/cvpops/test-operator:missing-default-channel-v1
-        /run_tests.py
-        """
-        result = subprocess.run(exec_cmd, shell=True)
-        self.assertFalse(os.path.exists(".errormessage"), "File .errormessage should not be present!")
+        self.env["IMAGE_TO_TEST"] = "quay.io/cvpops/test-operator:missing-default-channel-v1"
+        result = subprocess.run(self.exec_cmd,
+                                shell=True,
+                                env=self.env,
+                                cwd=SCRIPT_DIR)
         self.assertEqual(0, result.returncode)
 
     # Set env variable to custom made image that makes playbook extract-operator-bundle.yml fail
     # Due to parsing error
     def test_negative_parsing(self):
-        exec_cmd = """
-        export IMAGE_TO_TEST=quay.io/cvpops/test-operator:missing-alm-examples-v1
-        /run_tests.py
-        """
-        result = subprocess.run(exec_cmd, shell=True)
-        with open(".errormessage") as error_file:
+        self.env["IMAGE_TO_TEST"] = "quay.io/cvpops/test-operator:missing-alm-examples-v1"
+        result = subprocess.run(self.exec_cmd,
+                                shell=True,
+                                env=self.env,
+                                cwd=SCRIPT_DIR)
+        with open(SCRIPT_DIR+".errormessage") as error_file:
             self.assertIn("Result code:", error_file.read(), "Result code not found in %s" % error_file)
         self.assertEqual(50, result.returncode)
-        os.remove(".errormessage")
```

Comment:
Why it was removed? Each test should have clean "env" and leftover from previous test shoudl not cause failure to other test.

Reply:
Updated